recipes/rapidcheck/all/conanfile.py

```diff
@@ -103,15 +123,33 @@ def _module_file_rel_path(self):
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "rapidcheck"
         self.cpp_info.names["cmake_find_package_multi"] = "rapidcheck"
-        self.cpp_info.builddirs.append(self._module_subfolder)
-        self.cpp_info.build_modules["cmake_find_package"] = [self._module_file_rel_path]
-        self.cpp_info.build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
-        self.cpp_info.libs = ["rapidcheck"]
         
+        self.cpp_info.components["core"].set_property("cmake_target_name", "rapidcheck")
+        self.cpp_info.components["core"].builddirs.append(self._module_subfolder)
+        self.cpp_info.components["core"].set_property("cmake_build_modules", [self._module_file_rel_path])
+        self.cpp_info.components["core"].libs = ["rapidcheck"]
+        self.cpp_info.components["core"].includedirs  = ["include"]
         version = self.version[4:]
         if tools.Version(version) < "20201218":
             if self.options.enable_rtti:
-                self.cpp_info.defines.append("RC_USE_RTTI")
+                self.cpp_info.components["core"].defines.append("RC_USE_RTTI")
         else:
             if not self.options.enable_rtti:
-                self.cpp_info.defines.append("RC_DONT_USE_RTTI")
+                self.cpp_info.components["core"].defines.append("RC_DONT_USE_RTTI")
+                
+        if(self.options.enable_catch):
+            self.cpp_info.components["catch"].requires = ["core"]
+            self.cpp_info.components["catch"].includedirs  = ["include"]
+        if(self.options.enable_gmock):
+            self.cpp_info.components["gmock"].requires = ["core"]
+            self.cpp_info.components["gmock"].includedirs  = ["include"]
+        if(self.options.enable_gtest):
+            self.cpp_info.components["gtest"].requires = ["core"]
+            self.cpp_info.components["gtest"].includedirs  = ["include"]
+        if(self.options.enable_boost):
+            self.cpp_info.components["boost"].requires = ["core"]
+            self.cpp_info.components["boost"].includedirs  = ["include"]
+        if(self.options.enable_boost_test):
+            self.cpp_info.components["boost_test"].requires = ["core"]
+            self.cpp_info.components["boost_test"].includedirs  = ["include"]
```

Comment:
include is the default? https://docs.conan.io/en/latest/reference/conanfile/attributes.html#cpp-info

<html>
<body>
<!--StartFragment-->

NAME | DESCRIPTION
-- | --
self.cpp_info.includedirs | Ordered list with include paths. Defaulted to ["include"]

<!--EndFragment-->
</body>
</html>

```suggestion
        if(self.options.enable_catch):
            self.cpp_info.components["catch"].requires = ["core"]
        if(self.options.enable_gmock):
            self.cpp_info.components["gmock"].requires = ["core"]
        if(self.options.enable_gtest):
            self.cpp_info.components["gtest"].requires = ["core"]
        if(self.options.enable_boost):
            self.cpp_info.components["boost"].requires = ["core"]
        if(self.options.enable_boost_test):
            self.cpp_info.components["boost_test"].requires = ["core"]
```